admin_client.go

```diff
@@ -154,3 +158,75 @@ func (c *client) checkProcedureWithBackoff(ctx context.Context, procID uint64) e
 		}
 	}
 }
+
+// Snapshot creates a snapshot in HBase.
+//
+// Given that this operation takes some time to run, it's reccomended you
+// it in it's own channel without a context timeout. If a context timeout
+// is received, the client will not cleanup the snapshot you created.
+func (c *client) Snapshot(t *hrpc.Snapshot) error {
+	pbmsg, err := c.SendRPC(t)
+	if err != nil {
+		return err
+	}
+
+	_, ok := pbmsg.(*pb.SnapshotResponse)
+	if !ok {
+		return fmt.Errorf("sendPRC returned not a SnapshotResponse")
+
+	}
+
+	ticker := time.NewTicker(c.snapValidateInterval)
+	check := hrpc.NewCheckSnapshot(t)
+	ctx := t.Context()
+
+	for {
+		select {
+		case <-ticker.C:
+			pbmsgs, err := c.SendRPC(check)
+			if err != nil {
+				return err
+			}
+
+			r, ok := pbmsgs.(*pb.IsSnapshotDoneResponse)
+			if !ok {
+				return fmt.Errorf("sendPRC returned not a IsSnapshotDoneResponse")
+
+			}
+
+			if *r.Done {
+				return nil
+			}
+		case <-ctx.Done():
+			t.SetContext(context.Background())
+			err := c.DeleteSnapshot(t)
+			if err != nil {
+				return fmt.Errorf("failed to abort snapshot: %v", err)
+			}
+			return ctx.Err()
+		}
+	}
+}
+
+// DeleteSnapshot deletes a snapshot in HBase.
+//
+// Given that this operation takes some time to run, it's reccomended you
+// it in it's own channel without a context timeout. If a context timeout
+// is received, the client will not cleanup the snapshot you created.
+func (c *client) DeleteSnapshot(t *hrpc.Snapshot) error {
+	rt := hrpc.NewDeleteSnapshot(t)
+
+	pbmsg, err := c.SendRPC(rt)
+	if err != nil {
+		return err
+	}
+
+	_, ok := pbmsg.(*pb.DeleteSnapshotResponse)
+	if !ok {
+		return fmt.Errorf("sendPRC returned not a DeleteSnapshotResponse")
+
+	}
+
+	// TODO: It might be worth ensuring the snapshot no longer exists.
```

Comment:
It looks like DeleteSnapshotRequest is blocking and once it returns the response, the snapshot directory is removed (because they call post procedure hook): https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotManager.java#L340

so you don't need to check